File: cpp/examples/semantic_perception_registration.cpp
#include "registration_common.hpp"

int main() {
    auto set = examples::standard_fleet(
        {"atlas", "nova", "orion", "luna"},
        horus::core::RobotType::WHEELED,
        {0.8, 0.55, 0.45},
        "wheeled");
    examples::add_semantic_boxes(set.datavizs.front());
    return examples::register_or_fail(set);
}

File: rust/examples/semantic_perception_registration.rs
mod common;

use horus::{RobotDimensions, RobotType};

fn main() {
    let mut set = common::standard_fleet(
        &["atlas", "nova", "orion", "luna"],
        RobotType::Wheeled,
        RobotDimensions::new(0.8, 0.55, 0.45),
        "wheeled",
    );
    common::add_semantic_boxes(&mut set.datavizs[0]);
    common::register_or_fail(&mut set);
}
